src/main/groovy/com/bmuschko/gradle/clover/InstrumentCodeAction.groovy

```diff
@@ -310,7 +310,8 @@ class InstrumentCodeAction implements Action<Task> {
     private void compileGroovyAndJava(AntBuilder ant, Collection<File> srcDirs, File destDir, String classpath) {
         if (srcDirs.size() > 0) {
             String args = getAdditionalArgs()
-            ant.groovyc(destdir: destDir.canonicalPath, classpath: classpath, encoding: getEncoding()) {
+            Map groovycAttrs = [destdir: destDir.canonicalPath, classpath: classpath, encoding: getEncoding()] + (getAdditionalGroovycOpts() ?: [:])
+            ant.groovyc(groovycAttrs) {
```

Comment:
Simple, elegant. I will pull this in the 2.2.2 patch by the end of the week. Could you see if you can make a validation assertion that the added arguments did something in the functional test? Perhaps examining the task output?

Reply:
Yes, I'll see what I can do.